File: crates/normalize-facts/tests/fixtures/rust/add_numbers/project/src/math.rs
/// Returns the sum of `a` and `b`.
pub fn add(a: i32, b: i32) -> i32 {
    a + b
}

/// Returns the product of `a` and `b`.
pub fn multiply(a: i32, b: i32) -> i32 {
    a * b
}

/// Stateful calculator that records history.
pub struct Calculator {
    history: Vec<i32>,
}

impl Calculator {
    pub fn new() -> Self {
        Self { history: Vec::new() }
    }

    pub fn compute(&mut self, op: &str, a: i32, b: i32) -> i32 {
        let result = if op == "add" { add(a, b) } else { multiply(a, b) };
        self.history.push(result);
        result
    }
}

impl Default for Calculator {
    fn default() -> Self {
        Self::new()
    }
}

File: crates/normalize-facts/tests/fixtures/c/add_numbers/project/math.c
#include "math.h"
#include <string.h>

int add(int a, int b) {
    return a + b;
}

int multiply(int a, int b) {
    return a * b;
}

void calculator_init(Calculator *calc) {
    calc->count = 0;
}

int calculator_compute(Calculator *calc, const char *op, int a, int b) {
    int result;
    if (strcmp(op, "add") == 0) {
        result = add(a, b);
    } else {
        result = multiply(a, b);
    }
    calc->history[calc->count++] = result;
    return result;
}
